block-producer/src/txqueue/mod.rs

```diff
@@ -158,6 +160,8 @@ where
     TV: TransactionVerifier,
     BB: BatchBuilder,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(skip(self), ret, err(Debug), fields(COMPONENT))]
     async fn add_transaction(
         &self,
         tx: SharedProvenTx,
```

Comment:
A few comments here:

1. Logging full `SahredProvenTx` here would probably be too verbose. We can probably just log the transaction ID field instead of the full struct.
2. Do we need to log the return value since it is just `()`?
3. Similarly, do we need to log the error value? I'm assuming it would be captured by the parent span, but maybe not?
4. Do we also need log the `COMPONENT` here? Or would specifying it just for the parent span be enough?

Also, I think we should add an event at the and of this functions (i.e., line 175 below) - something like "Added transaction to proven transaction queue" with transaction ID and maybe current number of transactions in the queue.

Reply:
@bobbinth 
1. Agree, will log it manually.
2. You're right, I will remove it.
3. My preference is to log errors on each point, because code might be changed later and some errors might be processed/swallowed silently in parent calls.
4. Currently we use logging to console/file, which is not grouped by spans. In this case it's better to have `COMPONENT` label for each line.

> Also, I think we should add an event at the and of this functions (i.e., line 175 below) - something like "Added transaction to proven transaction queue" with transaction ID and maybe current number of transactions in the queue.

Got it!